### DISCUSSION TITLE
4.0.0.alpha1?

### DISCUSSION THREAD

**Discussion by jrochkind** [ElMassimo/vite_ruby#273] [Q&A] [answered]
I noticed there's a `4.0.0.alpha1` of both vite_ruby and vite_rails, both released in January 2022, 9 months ago.  No 4.x releases since then.  Neither one seems to show up in the CHANGELOG documents of the respective gem. 

What are the intentions behind when this goes 4.0.0 final? 

If starting a new project, would you recommend I use 4.0.0.alpha1 now, or the latest 3.x?

**Comment by ElMassimo** [comment, marked as answer]
Hi Jonathan!

> If starting a new project, would you recommend I use 4.0.0.alpha1 now, or the latest 3.x?

Latest stable release, `vite_ruby@3.2.4`.

---

> What are the intentions behind when this goes 4.0.0 final?

That alpha version includes an implementation of subresource integrity that can leverage my [`vite-plugin-manifest-sri`](https://github.com/ElMassimo/vite-plugin-manifest-sri) plugin, as mentioned in:

- https://github.com/ElMassimo/vite_ruby/issues/176

Since it requires breaking changes to the public API of `ViteRuby::Manifest`, and I'm not convinced that iteration is the right API, I'm waiting until more people request that functionality (or other factors, such as a breaking change in Vite's manifest format) prompt a major release.

Ideally, if nothing changes I'll sync the release of this breaking change with the release of Vite 4. Following the major version is completely arbitrary, but it does help people to prevent confusion.

**Comment by jrochkind** [comment]
OK thanks! Sounds like a new user should stick with 3.x for now. If someone else is wondering, maybe they can find this Q&A here now.
